### ISSUE TITLE
Misleading error message when accessing a property in violation of noPropertyAccessFromIndexSignature

### ISSUE THREAD

**Issue by bwbuchanan** [typed-ember/glint#512]
When the `noPropertyAccessFromIndexSignature` TypeScript compiler option is enabled and a template attempts a property access that fails this check, the error message is misleading.

Example:
```typescript

const record: Record<string, string> = {};

<template>
  {{record.foo}}
</template>
```

Error message:

```
error TS4111: Property 'foo' comes from an index signature, so it must be accessed with ['foo'].
```

It is not possible to write `{{record['foo']}}` in a handlebars template because this is not valid syntax, so the error message should be rewritten to suggest the `{{get}}` helper instead.
